cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/InMemoryProgramRunDispatcher.java

```diff
@@ -497,8 +505,21 @@ private ProgramOptions createPluginSnapshot(ProgramOptions options, ProgramId pr
 
     Set<String> files = Sets.newHashSet();
     HashMap<String, String> arguments = new HashMap<>(options.getArguments().asMap());
-    for (Map.Entry<String, Plugin> pluginEntry : appSpec.getPlugins().entrySet()) {
-      Plugin plugin = pluginEntry.getValue();
+    Hasher hasher = Hashing.sha256().newHasher();
+
+    /**
+     * If there is an artifact with SNAPSHOT version, hash should not be computed because artifact with
+     * SNAPSHOT might get changed but hash value remaining the same.
+     * {@link Constants.AppFabric.ARTIFACTS_SNAPSHOT_COMPUTE_HASH} allows to compute hash on snapshots
+     * which should only be used in testings.
+     */
+    boolean computeHash = cConf.getBoolean(Constants.AppFabric.ARTIFACTS_COMPUTE_HASH) &&
```

Comment:
nit: I think it would be a bit more readable if we lookup the cConf values in the constructor and store them in boolean vars. Would at least make the lines shorter.